blog/2017-02-03-using-native-driver-for-animated.md

```diff
@@ -0,0 +1,146 @@
+
+## What is this?
+
+The Animated API was designed with a very important constraint in mind, it is serializable. This means we can send everything about the animation to native before it has even started and allows native code to perform the animation on the UI thread without having to go through the bridge on every frame. It is very useful because once the animation has started, the JS thread can be blocked and the animation will still run smoothly. In practice this can happen a lot because user code runs on the JS thread and react renders can also lock JS for a long time.
+
+## A bit of history...
+
+This project started about a year ago, when Exponent built the li.st app on Android. [Krzysztof Magiera](https://twitter.com/kzzzf) was contracted to build the initial implementation on Android. It ended up working well and li.st was the first app to ship with native driven animations using Animated. A few months later [Brandon Withrow](https://github.com/buba447) built the initial implementation on iOS and after that [Ryan Gomba](https://twitter.com/ryangomba) and myself worked on adding missing features like support for `Animated.event` as well as squash bugs we found when using it in production apps. This was truly a community effort and I would like to thanks everyone that was involved as well as Exponent for sponsoring a large part of the development. As of today it is now used by `Touchable` components in React Native as well as for navigation animations in the newly released [React Navigation](https://github.com/react-community/react-navigation) library.
+
+## How does it work?
+
+First let's check out how animations currently work using Animated with the JS driver. When using Animated, you declare a graph of nodes that represent the animations that you want to perform, and then use a driver to update an Animated value using a predefined curve. You may also update an Animated value by connecting it to an event of a `View` using `Animated.event`.
+
+<INSERT COOL IMAGE?>
+
+Here's a breakdown of the steps for an animation and where it happens:
+- JS: The animation driver uses `requestAnimationFrame` to execute on every frame and update the value it drives using the new value it calculates based on the animation curve.
+- JS: Intermediate values are calculated and passed to a props node that is attached to a `View`.
+- JS: The `View` is updated using `setNativeProps`.
+- JS to Native bridge.
+- Native: The `UIView` or `android.View` is updated.
+
+As you can see most of the work happens on the JS thread so if it is blocked the animation will skip frames. It also needs to go through the JS to Native bridge on every frame to update native views.
+
+What the native driver does is since Animated produces a graph of animated nodes, it can be serialized and sent to native only once when the animation starts. After that native code can take care of updating the views directly on the UI thread on every frame.
```

Comment:
Possible alternative phrasing:

What the native driver does is move all of these steps to native. Since Animated produces a graph of animated nodes, it can be serialized and sent to native only once when the animation starts, eliminating the need to callback into the JS thread; the native code can take care of updating the views directly on the UI thread on every frame.